src/app/chat/[uuid]/page.tsx

```diff
@@ -1,23 +1,28 @@
 "use client";
 
 import { notFound, useParams } from "next/navigation";
+import { useEffect, useState } from "react";
 
+import { useChats } from "@/hooks/use-chats";
 import type { Chat } from "@/types/chat";
 
 export default function ConversationPage() {
   const parameters = useParams<{ uuid: string }>();
+  const { getChat } = useChats();
+  const [chat, setChat] = useState<Chat | null | undefined>(null);
 
-  // Get chats from local storage
-  const chats =
-    localStorage.getItem("chats") == null
-      ? []
-      : (JSON.parse(localStorage.getItem("chats")!) as Chat[]);
+  useEffect(() => {
+    const foundChat = getChat(parameters.uuid);
+    setChat(foundChat);
 
-  // Find chat by uuid
-  const chat = chats.find((c) => c.uuid === parameters.uuid);
+    //TODO call api there
```

Comment:
**suggestion**: With the current setup, if we implement api logic where that comment is, we will call the api on every chat page load, even if the chat already has a response from the model. We should check whether a response from the model is already saved within the chat object before calling the api. This whole section will be probably refactored as you resolve #31, but I wanted to point this out

Example:
```tsx
  useEffect(() => {
    const foundChat = getChat(parameters.uuid);
    setChat(foundChat);

    // or whatever condition that will signify that we already have a response
    if (!foundChat.recommendation) {
      // api logic here
    }
  }, [parameters.uuid, getChat]);
```

Reply:
yup, you're absolutely right